calico/felix/masq.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Metaswitch Networks 2015. All rights reserved.
+
+import logging
+from calico.felix.actor import Actor, actor_message
+from calico.felix.futils import IPV4, IPV6
+from calico.felix.ipsets import Ipset, FELIX_PFX
+
+_log = logging.getLogger(__name__)
+
+
+ALL_POOLS_SET_NAME = FELIX_PFX + "all-ipam-pools"
+MASQ_POOLS_SET_NAME = FELIX_PFX + "masq-ipam-pools"
+
+
+class MasqueradeManager(Actor):
+    def __init__(self, ip_type, iptables_mgr):
+        super(MasqueradeManager, self).__init__(qualifier=str(ip_type))
+        assert ip_type in (IPV4, IPV6)
```

Comment:
We only actually support IPv4 at present.  I don't think this will work with IPv6 anyway, so little point pretending like it might (the rule we install only uses one pair of ipsets, when I think we'd need v4 and v6 versions?).  I'm also not entirely sure if IPv6 NAT is enough of a thing that it even makes sense to do this.